cmd/openshift-install/internal_integration_test.go

```diff
@@ -219,6 +277,9 @@ func isoCmp(ts *testscript.TestScript, neg bool, args []string) {
 
 	workDir := ts.Getenv("WORK")
 	isoPath, aFilePath, eFilePath := args[0], args[1], args[2]
+	if skipPullSecretCmp(ts, aFilePath, eFilePath) {
```

Comment:
Should we move this down to line 296 so we are only skipping the byteCompare() (i.e. still check that the file exists in the place we expect it)?

Even then though, looking at some of the actual tests, we pass a pull secret like:

```
pullSecret: '{"auths": {"quay.io": {"auth": "c3VwZXItc2VjcmV0Cg=="}}}'
```

and in the output expect something like:

```
apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: null
  name: ostest-pull-secret
  namespace: cluster0
stringData:
  .dockerconfigjson: |-
    {
      "auths": {
        "quay.io": {
          "auth": "c3VwZXItc2VjcmV0Cg=="
        }
      }
    }
```

so there's a _lot_ we are missing by not checking the content.

Maybe we could take a similar approach to the release image and do something like:

```
pullSecret: '{"auths": {"quay.io": {"auth": "$QUAY_PULL_SECRET_AUTH"}}, {"ci.openshift.org": {"auth": "$OPENSHIFTCI_PULL_SECRET_AUTH"}}}'
```

(note - I don't actually know what the pull secret we get in CI looks like or how much of it we need.)

Then we could just set the values in the environment, wouldn't have to load-and-save the install-config, and could verify the full output format.

Reply:
So the case you were referring, for example, is https://github.com/openshift/installer/blob/master/cmd/openshift-install/testdata/agent/image/manifests/default_manifests.txt but here we are comparing the pull-secret with its contents correctly, isn't it?  e.g. in https://github.com/openshift/installer/blob/275af57895a88f9df7dc392ca87fc0d51418b1be/cmd/openshift-install/testdata/agent/image/manifests/default_manifests.txt#L11 we are checking the contents of `/etc/assisted/manifests/pull-secret.yaml` against `expected/pull-secret.yaml` ?

Reply:
Yeah, replacing all of that would be messy.
Presumably the only thing we use the pull secret for in the integration tests is to grab the ISO from the release image? If it turned out that a pull secret with only a couple of these auths was sufficient, then that starts to look more feasible.

> So the case you were referring, for example, is https://github.com/openshift/installer/blob/master/cmd/openshift-install/testdata/agent/image/manifests/default_manifests.txt but here we are comparing the pull-secret with its contents correctly, isn't it? e.g. in
> 
> [installer/cmd/openshift-install/testdata/agent/image/manifests/default_manifests.txt](https://github.com/openshift/installer/blob/275af57895a88f9df7dc392ca87fc0d51418b1be/cmd/openshift-install/testdata/agent/image/manifests/default_manifests.txt#L11)
> 
> Line 11 in [275af57](https://github.com/openshift/installer/commit/275af57895a88f9df7dc392ca87fc0d51418b1be)
>  isocmp agent.x86_64.iso /etc/assisted/manifests/pull-secret.yaml expected/pull-secret.yaml 
> we are checking the contents of /etc/assisted/manifests/pull-secret.yaml against expected/pull-secret.yaml ?

We were, but this change to `isoCmp()` means that we will no longer actually compare the contents in CI. So if any of the formatting around the pull secret changes, we will not be able to detect it.

Reply:
> So if any of the formatting around the pull secret changes, we will not be able to detect it.

The e2e jobs will fail in those case